framework/packages/abstract-client/tests/integration.rs

```diff
@@ -1552,3 +1570,42 @@ fn ans_balance() -> anyhow::Result<()> {
     assert_eq!(balance, Uint128::new(101));
     Ok(())
 }
+
+// Tests wether using the Account builder with install_on_subaccount on an existing account installs given apps
+#[test]
+fn account_fetcher_shouldnt_install_module_on_existing_account() -> anyhow::Result<()> {
+    let chain = MockBech32::new("mock");
+    let client = AbstractClient::builder(chain.clone()).build_mock()?;
+
+    let app_publisher: Publisher<MockBech32> = client
+        .account_builder()
+        .namespace(Namespace::new(TEST_WITH_DEP_NAMESPACE)?)
+        .build()?
+        .publisher()?;
+
+    let app_dependency_publisher: Publisher<MockBech32> = client
+        .account_builder()
+        .namespace(Namespace::new(TEST_NAMESPACE)?)
+        .build()?
+        .publisher()?;
+
+    app_dependency_publisher.publish_app::<MockAppI<_>>()?;
+    let res = app_publisher.publish_app::<MockAppWithDepI<_>>();
+    assert!(res.is_ok());
+
+    const NEW_NAMESPACE: &str = "new-namespace";
+
+    // We create an account
+    let account = client
+        .account_builder()
+        .namespace(NEW_NAMESPACE.try_into()?)
+        .build()?;
+
+    client.fetch_or_build(NEW_NAMESPACE.try_into()?, |builder| {
+        builder
+            .install_app::<MockAppWithDepI<MockBech32>>(&MockInitMsg {})
+            .unwrap()
```

Comment:
Yeah but in the case you don't want to install app for instance, you will have to return a wrapped Ok, wchich is very ugly ! So that's an oustading question, what do you think ?

Reply:
I would argue that `.unwrap()` is ugly as well! Also handling result is more convenient, than handling the panic

Reply:
One solution I had in mind, is to remove the Result from the install_app method, what do you think ?

Reply:
> One solution I had in mind, is to remove the Result from the install_app method, what do you think ?

Sure, we can move all of the checks inside `build()`

Reply:
Ah this is an idea yes. I was proposing to do unwraps inside the `AccountBuilder::install_app` instead but your suggestion makes sense

Reply:
I decided to go with `unwrap` inside `isntall_app` methods because : 
1. It's very difficult to handle the Vec<Result<ModuleInstallConfig, _>> because of clone bounds on errors mainly
2. It's ok to have that operation panic, because this is mainly parsing error and this is never on the blockchain

Reply:
> I decided to go with `unwrap` inside `isntall_app` methods because :
> 
>     1. It's very difficult to handle the Vec<Result<ModuleInstallConfig, _>> because of clone bounds on errors mainly
> 
>     2. It's ok to have that operation panic, because this is mainly parsing error and this is never on the blockchain

We can update and figure it out later if someone would be unhappy with this panic. Removing results from installs already nice!